src/createServer.js

```diff
@@ -1,3 +1,68 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+const createServer = () => {
+  return http.createServer((req, res) => {
+    const contentTypeHeader = { 'Content-Type': 'application/json' };
+
+    try {
+      const errors = [];
+
+      if (!req.url || req.url === '/') {
+        errors.push({
+          message: `Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".`,
+        });
+      }
+
+      const [path, queryString] = req.url.split('?') || [];
+      const params = new URLSearchParams(queryString);
+      const toCase = params.get('toCase');
+
+      if (!toCase) {
+        errors.push({
+          message: `"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".`,
+        });
+      }
+
+      const supportedCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+      if (toCase && !supportedCases.includes(toCase)) {
+        errors.push({
+          message: `This case is not supported. Available cases: ${supportedCases.join(', ')}.`,
+        });
+      }
+
+      const textToConvert = path?.slice(1);
+
+      if (!textToConvert && !req.url.startsWith('?')) {
```

Comment:
The condition `!textToConvert && !req.url.startsWith('?')` might not correctly identify when the text is missing. Since `textToConvert` is derived from `path.slice(1)`, it will be an empty string if the path is just '/'. Consider checking if `textToConvert` is an empty string instead of relying on `req.url.startsWith('?')`.